yaml/docker-compose/port-all-interfaces.yaml

```diff
@@ -0,0 +1,28 @@
+rules:
+  - id: port-all-interfaces
+    message: Service port is exposed on all interfaces
+    languages: [yaml]
+    severity: WARNING
+    metadata:
+      category: security
+      cwe: "CWE-1327: Binding to an Unrestricted IP Address"
+      subcategory: [audit]
+      technology: [docker, compose]
+      confidence: LOW
+      likelihood: LOW
+      impact: LOW
+      references:
+        - https://docs.docker.com/compose/compose-file/compose-file-v3/#ports
+    patterns:
+      - pattern-inside: |
+          services:
+            ...
+      - pattern: |
+          ports:
+            - ...
+            - "$PORT"
+            - ...
+      - focus-metavariable: $PORT
+      - metavariable-regex:
+          metavariable: $PORT
+          regex: "^(?!127.0.0.1:).+"
```

Comment:
Nitpick: Technically, 127.0.0.2, ..., 127.0.0.255, ..., 127.255.255.254 (127.255.255.255 will be a broadcast address afaik) will also point to your machine in most cases. Feel free to ignore :P

Reply:
Ahh, nice catch. I'll update the regex to catch those too 👍